### ISSUE TITLE
Phase 2: model lifecycle を実装する（candidate / metadata / quality gate / promotion）

### ISSUE THREAD

**Issue by yo4e** [yo4e/gaiden-sieve#4]
## 目的

Phase 1 で作った TF-IDF + Logistic Regression の学習・評価ループを、MLOps の「model lifecycle」として追跡・昇格できる形へ進める。

この Phase の主目的は、**モデルを学習して終わりにせず、「何から作られ、どう評価され、なぜ本番採用されたか」を辿れる状態にすること**。

```text
data + code
    ↓
  train
    ↓
candidate artifact
    ↓
metadata
    ↓
evaluate
    ↓
quality gate
 ├─ fail → reject
 └─ pass → explicit promote
              ↓
          production
```

## 実装範囲

`DESIGN.md` の Phase 2 に従い、以下を実装する。

- [ ] 学習済み scikit-learn `Pipeline` を joblib で candidate artifact として保存する
- [ ] profile ごとに candidate / production を分離する
- [ ] candidate に対応する metadata を保存する
- [ ] metadata に最低限、以下を記録する
  - [ ] profile
  - [ ] model version / artifact identifier
  - [ ] training data hash
  - [ ] code commit SHA（取得できる場合）
  - [ ] random seed / test size
  - [ ] 主要 hyperparameter
  - [ ] Precision / Recall / F1
  - [ ] created_at
- [ ] candidate を保存後に再ロードして評価できるようにする
- [ ] profile の `precision_gate` / `recall_gate` を使う quality gate を実装する
- [ ] gate 不通過の candidate は production に昇格できない
- [ ] promotion は明示操作にする
- [ ] promotion 後の production model をロードして1件分類できるようにする
- [ ] candidate / production / metadata の対応関係をテストする
- [ ] README に Phase 2 の学習ポイントを追記する

## 学習時に確認すること

1. **artifact とは何か**
   - 学習済みモデルも、コードとは別に管理すべき成果物である。

2. **metadata / data lineage はなぜ必要か**
   - 「このモデルが何のデータ・コード・設定から作られたか」を後から辿れるようにするため。

3. **candidate と production の違い**
   - 新しく学習できたことと、本番採用してよいことは別。

4. **quality gate は何を守るのか**
   - 新しいモデルが既存基準を下回ったまま production に入るのを防ぐ。

5. **なぜ promotion を自動化しすぎないのか**
   - v0 では評価と運用を観測できることを優先し、CIが勝手にproductionを書き換えない。

## 制約

- RSS 接続はまだしない
- GitHub Actions / drift / uncertain queue は Phase 3
- gold set の本格運用は後続検討
- 高度な Model Registry / cloud artifact store は使わない
- v0 の小さい artifact はリポジトリ内で扱ってよい
- `main` への直接実装はせず、専用ブランチ + PR で進める

## 完了条件

- [ ] `pytest` が通る
- [ ] fixture から candidate model を作成・保存できる
- [ ] candidate metadata から学習条件と評価結果を追跡できる
- [ ] quality gate の pass / fail を再現できる
- [ ] fail candidate は production に昇格できない
- [ ] pass candidate を明示的に production へ昇格できる
- [ ] production model をロードして新しい `title + summary` を分類できる
- [ ] Phase 3 の CI / reproducibility / drift へ進める状態になっている

## 正本

- `AGENTS.md`
- `DESIGN.md` §8, §9, §14, §16, §18, §19

Phase 1: #2
